identity/src/main/java/com/android/identity/credential/NameSpacedData.kt

```diff
@@ -0,0 +1,386 @@
+        fun putEntryString(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: String
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeString(value))
+        }
+
+        /**
+         * Encode the given value as `bstr` CBOR and adds it to the builder.
+         *
+         * @param nameSpaceName the name space name.
+         * @param dataElementName the data element name.
+         * @param value A byte string.
+         * @return the builder.
+         */
+        fun putEntryByteString(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: ByteArray
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeBytestring(value))
+        }
+
+        /**
+         * Encode the given value as an integer or unsigned integer and adds it to the builder.
+         *
+         * @param nameSpaceName the name space name.
+         * @param dataElementName the data element name.
+         * @param value The value, as a `long`.
+         * @return the builder.
+         */
+        fun putEntryNumber(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: Long
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeNumber(value))
+        }
+
+        /**
+         * Encode the given value as a boolean and adds it to the builder.
+         *
+         * @param nameSpaceName the name space name.
+         * @param dataElementName the data element name.
+         * @param value The value, as a `boolean`..
+         * @return the builder.
+         */
+        fun putEntryBoolean(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: Boolean
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeBoolean(value))
+        }
+
+        /**
+         * Builds a [NameSpacedData] from the builder
+         *
+         * @return a [NameSpacedData] instance.
+         */
+        fun build(): NameSpacedData {
+            return NameSpacedData(map)
+        }
+    }
+
+    companion object {
+        /**
+         * Creates a new [NameSpacedData] from encoded CBOR.
+         *
+         * @param encodedCbor CBOR encoded in the format described by [.encodeAsCbor].
+         * @return A [NameSpacedData].
+         * @throws IllegalArgumentException if the given data is not valid CBOR.
+         * @throws IllegalArgumentException if the given data does not confirm to the CDDL in
+         * [.encodeAsCbor].
+         */
+        @JvmStatic
+        fun fromEncodedCbor(encodedCbor: ByteArray): NameSpacedData {
+            return fromCbor(Util.cborDecode(encodedCbor))
+        }
+
+        private fun fromCbor(dataItem: DataItem): NameSpacedData {
+            val ret = LinkedHashMap<String, LinkedHashMap<String, ByteArray>>()
```

Comment:
this has to change to use `mutableMap`

`val ret = mutableMapOf<String, MutableMap<String, ByteArray>>()`